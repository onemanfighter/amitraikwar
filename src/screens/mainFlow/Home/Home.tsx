import { Box, Img, Text, useColorMode } from '@chakra-ui/react';
import { TypeWriterText } from '@components';
import { color } from 'framer-motion';
import { useTranslation } from 'react-i18next';

const Home = () => {
  const { t } = useTranslation();
  const { colorMode } = useColorMode();

  return (
    <Box
      flex={1}
      display="flex"
      flexDirection="column"
      alignItems="flex-start"
      justifyContent="flex-start"
    >
      <Box
        display="flex"
        flexDirection="row"
        justifyContent={'space-between'}
        width={'100%'}
      >
        <Box marginY={5} display="flex" flexDirection="column">
          <Text
            fontSize="3xl"
            fontWeight="bold"
            bgGradient={'linear(to-b, green.500, blue.500)'}
            bgClip={'text'}
            fontFamily={'monospace'}
          >
            {t('HomeScreen.heyMessage')}
          </Text>

          <Text
            fontSize="6xl"
            fontWeight="bold"
            bgGradient={'linear(to-b, green.500, blue.500)'}
            bgClip={'text'}
            fontFamily={'monospace'}
          >
            {t('HomeScreen.name')}
          </Text>
          <TypeWriterText
            fixedText=""
            textSequence={[
              { text: t('HomeScreen.whatIdoText.text0'), wait: 2000 },
              { text: t('HomeScreen.whatIdoText.text1'), wait: 2000 },
              { text: t('HomeScreen.whatIdoText.text2'), wait: 2000 },
              { text: t('HomeScreen.whatIdoText.text3'), wait: 2000 },
              { text: t('HomeScreen.whatIdoText.text4'), wait: 2000 },
            ]}
            styles={{
              fontSize: '2em',
              fontWeight: 'bold',
              textAlign: 'center',
              fontFamily: 'monospace',
              alignSelf: 'start',
            }}
          />
        </Box>
        <Box
          display="flex"
          flexDirection="column"
          alignItems="center"
          justifyContent="center"
          flex={1}
        >
          <Img
            width={300}
            height={300}
            src={require('../../../assets/images/Illustration.png')}
            alt="Illustration"
            loading="lazy"
          />
        </Box>
      </Box>
      <Box
        display="flex"
        width={'100%'}
        flexDirection="column"
        justifyContent="center"
        marginTop={5}
        bgGradient={
          colorMode === 'light'
            ? 'linear(to-r, green.200, blue.400)'
            : 'linear(to-r, green.800, blue.900)'
        }
        boxShadow={'lg'}
        padding={'4'}
        borderRadius={'lg'}
      >
        <Text fontSize="2xl" fontWeight="bold" fontFamily={'monospace'}>
          {t('HomeScreen.aboutMe.description')}
        </Text>
      </Box>
      <Box>{/* Add project preview carousal here here*/}</Box>
      <Box
        display="flex"
        width={'100%'}
        flexDirection="column"
        justifyContent="center"
        marginTop={5}
        bgGradient={
          colorMode === 'light'
            ? 'linear(to-r, green.200, blue.400)'
            : 'linear(to-r, green.800, blue.900)'
        }
        boxShadow={'lg'}
        padding={'4'}
        borderRadius={'lg'}
      >
        <Text
          fontSize="md"
          fontWeight="500"
          fontFamily={'monospace'}
          align={'center'}
        >
          {t('HomeScreen.copyrightText')}
        </Text>
      </Box>
    </Box>
  );
};

export default Home;
