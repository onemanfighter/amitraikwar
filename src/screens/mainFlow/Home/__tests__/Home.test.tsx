import { render } from '@testing-library/react';
import Home from '../Home';
import { ThemeProvider } from '@providers';

jest.mock('react-social-icons', () => ({
  SocialIcon: ({ url }: { url: string }) => <div>{url}</div>,
}));

describe('Home', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <Home />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
