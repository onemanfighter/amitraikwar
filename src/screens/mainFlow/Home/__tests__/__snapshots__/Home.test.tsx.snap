// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="light",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <div
    class="css-235qui"
  >
    <div
      class="css-15vhhhd"
    >
      <div
        class="css-11d69m5"
      >
        <p
          class="chakra-text css-3t2enw"
        >
          Hey there! My name is
        </p>
        <p
          class="chakra-text css-k99qrb"
        >
          Amit Raikwar
        </p>
        <span
          class="index-module_type__E-SaG"
          style="font-size: 2em; font-weight: bold; text-align: center; font-family: monospace; align-self: start;"
        />
      </div>
      <div
        class="css-3zuqv9"
      >
        <img
          alt="Illustration"
          class="chakra-image css-rdnx5m"
          loading="lazy"
          src="[object Object]"
        />
      </div>
    </div>
    <div
      class="css-m38xl5"
    >
      <p
        class="chakra-text css-5h1wpx"
      >
        I am a Software Engineer with 3 years of experience in building scalableðŸš€ and reliableðŸŽ¯ software applications. I am passionate about building products that solve real-life problems and have a positive impact on society. I am a quick learner and always eager to learn new technologies and tools. I am a team player and love to work in a collaborative environment.
      </p>
    </div>
    <div
      class="css-0"
    />
    <div
      class="css-m38xl5"
    >
      <p
        class="chakra-text css-1h1lvy8"
      >
        Â© 2024 Amit Raikwar. All rights reserved. Designed and developed by Amit Raikwar.
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
